clang-tools-extra/clangd/index/SymbolCollector.cpp

```diff
@@ -821,7 +821,8 @@ void SymbolCollector::setIncludeLocation(const Symbol &S, SourceLocation DefLoc,
 
   // Use the expansion location to get the #include header since this is
   // where the symbol is exposed.
-  IncludeFiles[S.ID] = SM.getDecomposedExpansionLoc(DefLoc).first;
+  if (FileID FID = SM.getDecomposedExpansionLoc(DefLoc).first; FID.isValid())
```

Comment:
i still think it's wasteful to check both in both places, but now that you do a `lookup` instead of `at` below, i'll leave this to you.